.github/workflows/build.yml

```diff
@@ -0,0 +1,111 @@
+name: Publish wheels
+
+on:
+  push:
+    tags:
+      - v*
+  pull_request:
+    paths:
+      - 'Cargo.toml'
+      - 'pyproject.toml'
+
+jobs:
+  build:
+    # This workflow comes from https://github.com/pydantic/pydantic-core/blob/main/.github/workflows/ci.yml
+    name: build on ${{ matrix.os }} (${{ matrix.target }} - ${{ matrix.manylinux }})
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
This is a lot of config! 😅. It's not the prettiest, but I suppose this is the cleanest way to target all of these mutually exclusive options.